Content.Server/_DV/CosmicCult/Abilities/CosmicTransmuteSystem.cs

```diff
@@ -40,18 +40,19 @@ private void OnTransmuteWeaponGlyph(Entity<CosmicGlyphTransmuteWeaponComponent>
             return;
         }
 
-        Spawn(_random.Pick(uid.Comp.TransmuteWeapon), tgtpos);
+        Spawn(_random.Pick(uid.Comp.Transmutations), tgtpos);
         QueueDel(possibleTargets.First());
     }
 
+
     /// <summary>
-    ///     Gets all sharp items near a glyph.
+    ///     Gets all whitelisted entities near a glyph.
     /// </summary>
-    private HashSet<EntityUid> GatherSharpItems(EntityUid uid, float range)
+    private HashSet<EntityUid> GatherEntities(Entity<CosmicGlyphTransmuteComponent> ent)
     {
-        _sharpItems.Clear();
-        _lookup.GetEntitiesInRange(Transform(uid).Coordinates, range, _sharpItems, LookupFlags.Uncontained);
-        _sharpItems.RemoveWhere(item => !HasComp<SharpComponent>(item));
-        return _sharpItems;
+        _entities.Clear();
+        _lookup.GetEntitiesInRange(Transform(ent).Coordinates, ent.Comp.TransmuteRange, _entities);
```

Comment:
still need to use the one taking Type so its not checking every entity

Reply:
is this premature optimization or does iterating through all entities in a 1.5 radius as a one-off action actually kill performance that much

Reply:
premature but it also saves you having to use a whitelist for a single component when theres a dedicated thing for it

Reply:
armor transmute needs two components

Reply:
then have it be optional for that, like how proximity beeper works